### ISSUE TITLE
`no-this-in-sfc`: false-positives in non-React class

### ISSUE THREAD

**Issue by mrtnzlml** [yannickcr/eslint-plugin-react#3063]
Hello, we are getting new false-positive errors after upgrading to the latest version 7.25.1:

```text
error: Stateless functional components should not use `this` (react/no-this-in-sfc) at src/monorepo-shipit/src/ShipitConfig.js:70:47:
  68 |     return (changeset: Changeset) => {
  69 |       const ch1 = addTrackingData(changeset);
> 70 |       const ch2 = stripExceptDirectories(ch1, this.getSourceRoots());
     |                                               ^
  71 |       const ch3 = moveDirectories(ch2, this.directoryMapping);
  72 |       const ch4 = stripPaths(ch3, this.strippedFiles);
  73 | 


error: Stateless functional components should not use `this` (react/no-this-in-sfc) at src/monorepo-shipit/src/ShipitConfig.js:71:40:
  69 |       const ch1 = addTrackingData(changeset);
  70 |       const ch2 = stripExceptDirectories(ch1, this.getSourceRoots());
> 71 |       const ch3 = moveDirectories(ch2, this.directoryMapping);
     |                                        ^
  72 |       const ch4 = stripPaths(ch3, this.strippedFiles);
  73 | 
  74 |       // First we comment out lines marked with `@x-shipit-disable`.


error: Stateless functional components should not use `this` (react/no-this-in-sfc) at src/monorepo-shipit/src/ShipitConfig.js:72:35:
  70 |       const ch2 = stripExceptDirectories(ch1, this.getSourceRoots());
  71 |       const ch3 = moveDirectories(ch2, this.directoryMapping);
> 72 |       const ch4 = stripPaths(ch3, this.strippedFiles);
     |                                   ^
  73 | 
  74 |       // First we comment out lines marked with `@x-shipit-disable`.
  75 |       const ch5 = commentLines(ch4, '@x-shipit-disable', '//', null);


3 errors found.
```

The thing is that this class doesn't have anything to do with React or stateless functional components. You can inspect it yourself here: https://github.com/adeira/universe/blob/f7b9241c0fb7caa527c4d82e80f9255da58282aa/src/monorepo-shipit/src/ShipitConfig.js

I think this is a bug.

Thanks for having a look! 😎

**Comment by ljharb** [comment]
cc @Wesitos

**Comment by mrtnzlml** [comment]
Seems like this got fixed in 7.25.2 but not sure which commit exactly. Maybe https://github.com/yannickcr/eslint-plugin-react/commit/1a5a970f8a7a57680eed29150505309f5597d25c? Closing. 🙂
